### PR TITLE
docs: adopt Collective Funds Guidelines 0.1

### PR DESCRIPTION
**Pull request by voxpelli**
These guidelines were created by us Mocha maintainers with intent of being used for Mocha.

When #5197 is merged, do a follow up that adds the collective funds guidelines badge to the README.

### CODE REVIEW

**Review by Uzlopak — APPROVED** [review]
Great work. LGTM!

File: FUNDING.md
Comment (JoshuaKGoldberg) [inline FUNDING.md:1]: One note from #5027: 

> - [ ] Publicly propose and get approved an invoicing system for https://opencollective.com/mochajs, then reach out to previous maintainers to compensate them for their time

I don't know how long would be appropriate to leave this open. IMO the Collective Funds Guidelines being a shared public resource that's gone through feedback+review from several projects-worth of open source maintainers is enough for me. ✅

**Review by JoshuaKGoldberg — APPROVED** [review]
I love how simple this document is 😄 big ups from me on the Collective Funds Guidelines work. Awesome job @voxpelli! 👏

File: FUNDING.md
Comment (voxpelli) [reply FUNDING.md:1]: Yeah, I think so as well 👍

**Pull request merged**